### ISSUE TITLE
sanitycheck/CI is not building with all supported toolchains

### ISSUE THREAD

**Issue by zephyrbot** [zephyrproject-rtos/infrastructure-old#61]
**_Reported by Andrew Boie:_**
We are not building against all of our supported toolchains. This is a huge problem as any differences between these toold and vanilla GCC will not be detected until it is a fire drill, CI is not covering it. There could be breakage already. We need to fix this ASAP.

> Kumar Gala has posted comments on this change.

> 
Change subject: nanokernel: support GCC compiler atomic builtins
> ......................................................................

> 
> 
Patch Set 2:

> 
> Sanitycheck is passing with this patch.
>  >
>  > If we are supposedly supporting toolchains that aren't being built
>  > with as part of our continuous integration, that is a serious
>  > problem.

> 
We support the ARM embedded GCC (https://launchpad.net/gcc-arm-embedded), while similar to the Zephyr GNU tools, there are some subtle differences that I'm trying to work with the guys to resolve.

> 
I don't think we are having CI run for the toolchain (gccarmemb, issm, xtools)

(Imported from Jira INF-60)

**Comment by zephyrbot** [comment]
**_by Javier B Perez:_**

blocked by https://jira.zephyrproject.org/browse/ZEP-592

**Comment by zephyrbot** [comment]
**_by William  Mills:_**

newlib configuration in particular seems to break from toolchain to toolchain.


The main make files strips out the toolchains knowledge of where the headers and lib are and then subst its own knowledge.  Some of this can be compensated for in scripts/Makefile.toolchain.xyz but that is awkward and not hooked into configuration.


tests/kernel/test_build/  with CONF_FILE=newlib.conf is the only test that uses newlib and it is a build only test and only a nanokernel config.

**Comment by zephyrbot** [comment]
**_by Javier B Perez:_**

Since the required changes for the script were merged I will close this ticket.
If we get approval from TSC to start testing more toolchains in CI we can open a new request.
Note: more toolchains means more tests to build = increase the build time.

**Comment by zephyrbot** [comment]
**_by Sharron LIU:_**

Please reporter verify this

**Comment by zephyrbot** [comment]
**_by Mark Linkmeyer:_**

Moving to Closed since it was previously marked Resolved as Won't Do and got moved into the Merged state during the migration to the new Jira workflow.
